stock_move_line_removal_date/models/stock_move_line.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from datetime import datetime
+
+import pytz
+from dateutil.relativedelta import relativedelta
+from odoo import SUPERUSER_ID, fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    removal_date = fields.Date("Removal Date")
+
+    def _action_done(self):
+        res = super(StockMoveLine, self)._action_done()
+        for ml in self:
+            if ml.picking_id.picking_type_code == "incoming" and ml.removal_date:
+                removal_date = fields.Datetime.to_string(
+                    ml.convert_date_to_datetime(ml.sudo().removal_date, 12)
```

Comment:
Why we need sudo() here? If so, shouldn't we add `sudo()` in L19 as well? i.e. `if ml.picking_id.picking_type_code == "incoming" and ml.sudo().removal_date:`

Reply:
No need to add `sudo()`. I think this is a remnant of my attempt to use context_timestamp. I will remove it. Thanks for catching this!